plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OriginalFilesUtils.java

```diff
@@ -54,7 +54,7 @@ public static long getPrecedingRowCount(
         long rowCount = 0;
         for (OriginalFileInfo originalFileInfo : originalFileInfos) {
             if (originalFileInfo.getName().compareTo(splitPath.fileName()) < 0) {
-                Location path = splitPath.parentDirectory().appendPath(originalFileInfo.getName());
+                Location path = splitPath.replaceFilename(originalFileInfo.getName());
```

Comment:
Let's move this to separate commit, so that Delta fix comes with its test changes only and other cleanup (like this one + same in  `io.trino.plugin.deltalake.transactionlog.writer.S3NativeTransactionLogSynchronizer#write` and `HiveWriterFactory`) come as a second commit.

Reply:
Follow-up `HiveWriterFactory` https://github.com/trinodb/trino/pull/18183